cleanlab/rank.py

```diff
@@ -109,57 +123,335 @@ def get_self_confidence_for_each_label(
 
     labels : np.array
-      A discrete vector of noisy labels, i.e. some labels may be erroneous.
-      *Format requirements*: for dataset with K classes, labels must be in {0,1,...,K-1}.
+      Labels in the same format expected by the `score_label_quality()` method.
 
     pred_probs : np.array (shape (N, K))
-      P(label=k|x) is a matrix with K model-predicted probabilities.
-      Each row of this matrix corresponds to an example x and contains the model-predicted
-      probabilities that x belongs to each possible class.
-      The columns must be ordered such that these probabilities correspond to class 0,1,2,...
-      `pred_probs` should have been computed using 3 (or higher) fold cross-validation.
+      Predicted-probabilities in the same format expected by the `score_label_quality()` method.
 
     Returns
     -------
-      normalized_margin : np.array (float)
-        Return a score (between 0 and 1) for each example of its likelihood of
-        being correctly labeled. Assumes pred_probs is computed holdout/out-of-sample.
-        normalized_margin = prob_label - max_prob_not_label"""
+    label_quality_scores : np.array (float)
+      Return a score (between 0 and 1) for each example of its likelihood of
+      being correctly labeled.
+      normalized_margin = prob_label - max_prob_not_label
+
+    """
 
     self_confidence = get_self_confidence_for_each_label(labels, pred_probs)
     max_prob_not_label = np.array(
         [max(np.delete(pred_probs[i], l, -1)) for i, l in enumerate(labels)]
     )
     label_quality_scores = (self_confidence - max_prob_not_label + 1) / 2
     return label_quality_scores
+
+
+def get_confidence_weighted_entropy_for_each_label(
+    labels: np.array, pred_probs: np.array
+) -> np.array:
+    """Returns the "confidence weighted entropy" label-quality score for each datapoint.
+
+    This is a function to compute label-quality scores for classification datasets,
+    where lower scores indicate labels less likely to be correct.
+
+    "confidence weighted entropy" is the normalized entropy divided by "self-confidence".
+
+    Parameters
+    ----------
+    labels : np.array
+      Labels in the same format expected by the `score_label_quality()` method.
+
+    pred_probs : np.array (shape (N, K))
+      Predicted-probabilities in the same format expected by the `score_label_quality()` method.
+
+    Returns
+    -------
+    label_quality_scores : np.array (float)
+      Return a score (between 0 and 1) for each example of its likelihood of
+      being correctly labeled.
+
+    """
+
+    self_confidence = get_self_confidence_for_each_label(labels, pred_probs)
+
+    # Divide entropy by self confidence
+    label_quality_scores = get_entropy(**{"pred_probs": pred_probs}) / self_confidence
+
+    # Rescale
+    label_quality_scores = np.log(label_quality_scores + 1) / label_quality_scores
+
+    return label_quality_scores
+
+
+def get_entropy(pred_probs: np.array) -> np.array:
```

Comment:
For this function's return value, we currently have: higher values = more concerning, whereas for the label-quality scores, we have lower values = more concerning.  I think we need to standardize everything around lower = more concerning, so could change this score to `1 - normalized_entropy`.

What do you think?

Reply:
I think we should leave it as-is because entropy here is not used as a label quality score.

If we add it to the label quality scoring function in the future, we can rename it to "inv_entropy" or something similar and do the 1 - normalized_entropy.

Reply:
I guess because this function is currently merely intended as a helper function for `get_confidence_weighted_entropy_for_each_label`?

If so, maybe we move it to `utils/` as well?
I think for all user-facing scoring functions, we want to be unified and always have: lower = more concerning. 

So later when we introduce a scoring function based on this, we could add the `1 - entropy()` in that function and have it import this as a helper from `utils/` as well.

Reply:
Right now it's more of a helper function for confidence_weighted_entropy. We can consider putting it in label_quality_utils.py

Reply:
Just saw your previous comment. I think we commented at the same time :)

Agree, will move it to utils and we can add 1 - entropy() later if we introduce this as a scoring function

Reply:
Done, thanks
3dbbe7054aeae25b8bddd73a53838cce35b06fa9